pkg/ipam/ipam.go

```diff
@@ -210,7 +211,12 @@ func (ipam *IPAM) isIPExcluded(ip net.IP, pool Pool) (string, bool) {
 // PoolOrDefault returns the default pool if no pool is specified.
 func PoolOrDefault(pool string) Pool {
 	if pool == "" {
-		return PoolDefault
+		return PoolDefault()
 	}
 	return Pool(pool)
 }
+
+// PoolDefault returns the default pool
+func PoolDefault() Pool {
+	return Pool(option.Config.IPAMDefaultIPPool)
```

Comment:
What happens if a default pool is specified that doesn't exist?

Reply:
You will receive the following information, and the Pod cannot be created.
```
unable to allocate from pool test (family ipv4): pool not (yet) available
```

Reply:
Presumably that would happen on the first allocation attempt - in this case I wonder if it would be preferable to validate this early (@gandro)?

Reply:
I have not yet tested it with this PR, but basically cilium-agent should not be come ready if the default pool (or more precise, any pool from which we want to pre-allocate) does not exist. This is similar to how it works with e.g. a misconfigured ENI mode (e.g. the local node requests pod IPs from a ENI subnet that does not exist)


https://github.com/cilium/cilium/blob/d6b1637880da13ab76c8465851c08c630e93f10c/pkg/ipam/multipool.go#L328-L330